src/qe/mbp/mbp_arrays.cpp

```diff
@@ -726,7 +726,7 @@ namespace mbp {
             model_evaluator mev(mdl);
             mev.set_model_completion(true);
             M = &mdl;
-            m_mev = &mev;
+            flet<model_evaluator*> _mev(m_mev, &mev);
```

Comment:
It may be more dilligent but it doesn't address any bug.

Reply:
`mev` is allocated on stack inside this function. So, keep it as pointer inside `m_mev` lead to broken stack. If you wish I may go back to dynamic allocation of this, but using flet seems cleaner.

Reply:
Note that there is only one public method, m_mev doesn't get used from other methods.
This is why I say it doesn't address any existing bug. So it is more about cosmetics and better quality code. Any new code should have used the flet pattern, for old code there is a tradeoff between the overhead of churn.

Reply:
Well, `m_mev` is used inside this file outside of function on which stack it is allocated and from which stack a pointer has been get. If I understand code corectly, it keep pointer to `mev` allocated on the stack of `operator()`. When it's called who knows that value is here.

Have I missed something?

Reply:
There is no bug.
The only place the private class array_project_eqs_util is used is here:

```
        array_project_eqs_util pe (m);
        pe (mdl, arr_vars, fml, aux_vars);
        
```

There could be a future bug. But then there can be future bugs in many things.
There is risk even when adding fixes for future bugs.

Since we are spending time on this code, go ahead and remove the other two files that should not be changed.
The change is for the better, but the overhead in reviewing it and taking risk isn't the best use of time.

Reply:
anyway I can reproduce https://github.com/Z3Prover/z3/issues/6902 quite easy and I simple hunting to cause of this issue.